### PR TITLE
Create UrlForm wrapper and add EntityCodec

### PR DESCRIPTION
**Pull request by julien-truffaut**
All previous tests are passing but the scalacheck test I added is failing, I will look into it.

### CODE REVIEW

**Comment by julien-truffaut** [comment]
Sorry I have been busy, I am planning to come back to this issue

**Comment by bryce-anderson** [comment]
No worries, we all have work to do.

**Comment by rossabaker** [comment]
This one is real close. Do we want to squeeze it into 0.6 or defer?

**Comment by julien-truffaut** [comment]
@rossabaker we should close it, I don't have enough time to work on this at the moment. Sorry for the noise.

**Comment by rossabaker** [comment]
I think there's good work here.  I'll put it in the freezer so it doesn't get forgotten, and whoever can pick it back up.

**Comment by bryce-anderson** [comment]
What else do we want done on this? It seems alright to me, but others will be better at spotting papercuts. Its largely the same with the tests made to work without defining an arbitrary string satisfying an arbitrary charset.

File: core/src/test/scala/org/http4s/EntityDecoderSpec.scala [hunk 1]
```diff
@@ -59,26 +60,26 @@ class EntityDecoderSpec extends Http4sSpec {
...
-      getBody(resp.body) must_== "Jonathan Doe".getBytes
+      UrlForm.entityDecoder.decode(resp).run.run must_== \/-(urlForm)
     }
 
-    "handle a parse failure" in {
-      val body = strBody("%C")
-      val resp = server(Request(body = body)).run
-      resp.status must_== Status.BadRequest
-    }
+    // TODO: need to make urlDecode strict
```
Comment (rossabaker) [inline core/src/test/scala/org/http4s/EntityDecoderSpec.scala]: This is a good use of pendingUntilFixed.
https://coderwall.com/p/rgbj2a/use-pendinguntilfixed-to-disable-specs-temporarily

**Comment by rossabaker** [comment]
LGTM.  Arbitrary charset is recorded in #145.  More thoughtful parsing is covered in #194.  I think that pendingUntilFixed would be a good thing to do, and let's land it.

**Comment by julien-truffaut** [comment]
looks good to me, thanks @bryce-anderson for picking this up

**Pull request merged**